generator/net_snmp.go

```diff
@@ -157,24 +157,48 @@ var (
 // Warning: This function plays with the stderr file descriptor.
 func initSNMP(logger log.Logger) (string, error) {
 	// Load all the MIBs.
-	os.Setenv("MIBS", "ALL")
-	// Help the user find their MIB directories.
-	level.Info(logger).Log("msg", "Loading MIBs", "from", C.GoString(C.netsnmp_get_mib_directory()))
+	err := os.Setenv("MIBS", "ALL")
+	if err != nil {
+		return "", err
+	}
+	if *userMibsDir != "" {
+		// The user told us where to load the mibs
+		err = level.Info(logger).Log("msg", "Loading MIBs", "from", *userMibsDir)
+		if err != nil {
+			return "", err
+		}
+		C.netsnmp_set_mib_directory(C.CString(*userMibsDir))
+	} else {
+		// Help the user find their MIB directories.
+		err = level.Info(logger).Log("msg", "Loading MIBs", "from", C.GoString(C.netsnmp_get_mib_directory()))
```

Comment:
We don't need to worry about `err` from the logger. The error as actually never tripped due to the way the library works. This is why we ignore this in the golanci-lint config in the project.

Also, do we need to duplicate the log lines? Shouldn't `C.netsnmp_get_mib_directory()` return the correct thing after `C.netsnmp_set_mib_directory()`?

Reply:
> The error as actually never tripped due to the way the library works.

:facepalm: Thanks! I'll fix those places where I somewhat mindlessly checked for the err.

Reply:
>  Shouldn't `C.netsnmp_get_mib_directory()` return the correct thing after `C.netsnmp_set_mib_directory()`?

Hmm, I'll check that. I would expect that it should.